sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -219,11 +251,26 @@ static az_result started(az_event_policy* me, az_event event)
             (_az_hfsm*)me, (az_event){ .type = AZ_HFSM_EVENT_ERROR, .data = NULL }));
       }
 
-      _az_RETURN_IF_FAILED(_az_hfsm_transition_peer((_az_hfsm*)me, started, idle));
+      _az_FAULTED_IF_FAILED(
+          _az_hfsm_transition_peer((_az_hfsm*)me, started, idle), this_policy, started);
       _az_RETURN_IF_FAILED(
           az_event_policy_send_inbound_event((az_event_policy*)this_policy, event));
       break;
 
+    case AZ_HFSM_EVENT_ERROR:
+    case AZ_HFSM_EVENT_TIMEOUT:
+      if (event.data == &this_policy->_internal.connection_timer)
+      {
+        _az_RETURN_IF_FAILED(_az_hfsm_transition_peer((_az_hfsm*)me, started, faulted));
+        _az_RETURN_IF_FAILED(az_event_policy_send_inbound_event(
```

Comment:
Is this going to send two AZ_HFSM_EVENT_ERROR events? (one here and one on entrance to faulted?)

Reply:
This will do that, changed faulted to not send event on entry. When doing a transition to faulted made sure we always send an inbound event